plugins/azure/kubernetesservice/aksAgentVersion.js

```diff
@@ -0,0 +1,63 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+var compareVersions = require('compare-versions');
+
+module.exports = {
+    title: 'Kubernetes Latest Version',
+    category: 'Kubernetes Service',
+    description: 'Ensures the kubernetes version is same across the node pools with the cluster.',
+    more_info: 'AKS supports provisioning clusters from several versions of Kubernetes. Node pools should be at per with the cluster kubernetes version.',
+    recommended_action: 'Upgrade the version of Kubernetes on all AKS clusters node pool to the same version as the cluster.',
+    link: 'https://docs.microsoft.com/en-us/azure/aks/aad-integration',
+    apis: ['managedClusters:list'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.managedClusters, function(location, rcb) {
+            var managedClusters = helpers.addSource(cache, source,
+                ['managedClusters', 'list', location]);
+
+            if (!managedClusters) return rcb();
+
+            if (managedClusters.err || !managedClusters.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for Kubernetes clusters: ' + helpers.addError(managedClusters), location);
+                return rcb();
+            }
+
+            if (!managedClusters.data.length) {
+                helpers.addResult(results, 0, 'No existing Kubernetes clusters found', location);
+                return rcb();
+            }
+
+            managedClusters.data.forEach(managedCluster => {
+                var kubernetesVersion = managedCluster.kubernetesVersion;
+                var agentPoolProfiles = managedCluster.agentPoolProfiles;
+
+                if (!agentPoolProfiles || !agentPoolProfiles.length) {
+                    helpers.addResult(results, 3,
+                        'Unable to query for Kubernetes cluster node profile' , location, managedCluster.id);
+                } else {
+                    agentPoolProfiles.forEach(agentPoolProfile =>{
+                        if (agentPoolProfile.orchestratorVersion &&
+                            compareVersions(agentPoolProfile.orchestratorVersion,kubernetesVersion) === -1 ) {
```

Comment:
Can you use the same comparison logic from the `aksVersion` plugin to avoid adding another package dependency?
(We try to avoid new packages unless they are absolutely required).

Reply:
in aksversion we are just checking if there is any entry in updateprofile array to decide whether we are at latest version. We are not checking any version.But here we are checking version of the node pool with the aks cluster it self.

Reply:
Got it - is it possible to extract the relevant functionality from the package and put it in `helpers/azure.js` instead? Want to avoid adding any dependencies if we can.

Reply:
I have wrote a simple function for this and added in helpers/shared.js as this is not specific to azure. Please take a look